app/models/attachment_reference.rb

```diff
@@ -0,0 +1,57 @@
+class AttachmentReference < ActiveRecord::Base
+  after_initialize :set_expiry_at, if: :new_record?
+  before_save :update_expiry_at
+
+  belongs_to :attachable, polymorphic: true, inverse_of: :attachment_references
```

Comment:
I'm not sure, since the framework is attachable, maybe attachable is better ? and it also follows rails' polymorphic conversion.